source/state_representation/test/tests/test_trajectory.cpp

```diff
@@ -1,129 +1,377 @@
+
+  using PointType = typename std::conditional<
+      std::is_same_v<TypeParam, TrajectoryBaseInterface>, TrajectoryPoint,
+      typename std::conditional<std::is_same_v<CartesianTrajectory, TypeParam>, CartesianState, JointState>::type>::
+      type;
+
+  PointType point0;
+  PointType point1;
+  PointType point2;
+
+  if constexpr (std::is_same_v<PointType, TrajectoryPoint>) {
+    point0.data = Eigen::VectorXd::Random(3);
+    point0.duration = std::chrono::nanoseconds(10);
+    point1.data = Eigen::VectorXd::Random(3);
+    point1.duration = std::chrono::nanoseconds(20);
+    point2.data = Eigen::VectorXd::Random(3);
+    point2.duration = std::chrono::nanoseconds(30);
+  } else if constexpr (std::is_same_v<PointType, CartesianState>) {
+    point0 = CartesianState::Random("foo");
+    point1 = CartesianState::Random("bar");
+    point2 = CartesianState::Random("baz");
+  } else if constexpr (std::is_same_v<PointType, JointState>) {
+    point0 = JointState::Random("foo", 25);
+    point1 = JointState::Random("bar", 25);
+    point2 = JointState::Random("baz", 25);
+    this->trajectory->set_joint_names(point0.get_names());
+  }
+
+  std::vector<PointType> points = {point0, point1, point2};
+  std::vector<std::chrono::nanoseconds> durations = {
+      std::chrono::nanoseconds(10), std::chrono::nanoseconds(20), std::chrono::nanoseconds(30)
+  };
+  EXPECT_NO_THROW(this->add_points(points, durations));
+  ASSERT_FALSE(this->trajectory->is_empty());
+  auto trajectory_points = this->trajectory->get_points();
+  auto trajectory_durations = this->trajectory->get_durations();
+  for (unsigned int i = 0; i < this->trajectory->get_size(); ++i) {
+    EXPECT_NO_THROW(this->expect_equal(points[i], i, durations[i]));
+    EXPECT_NO_THROW(this->expect_equal(trajectory_points[i], i, durations[i]));
+    EXPECT_NO_THROW(this->expect_equal(this->trajectory->get_point(i), i, trajectory_durations[i]));
+  }
+
+  auto times_from_start = this->trajectory->get_times_from_start();
+  std::chrono::nanoseconds time_from_start = std::chrono::nanoseconds(0);
+  for (unsigned int i = 0; i < this->trajectory->get_size(); ++i) {
+    time_from_start += durations[i];
+    EXPECT_EQ(times_from_start[i], time_from_start);
+    EXPECT_EQ(this->trajectory->get_time_from_start(i), time_from_start);
+    EXPECT_EQ(this->trajectory->get_duration(i), durations[i]);
+  }
+  EXPECT_EQ(this->trajectory->get_trajectory_duration(), time_from_start);
+
+  this->trajectory->reset();
+  EXPECT_TRUE(this->trajectory->is_empty());
+  EXPECT_THROW(this->trajectory->get_size(), exceptions::EmptyStateException);
+  EXPECT_THROW(this->trajectory->get_durations(), exceptions::EmptyStateException);
+  if constexpr (std::is_same_v<PointType, CartesianState>) {
+    EXPECT_STREQ(this->trajectory->get_reference_frame().c_str(), "world");
+  } else if constexpr (std::is_same_v<PointType, JointState>) {
+    EXPECT_NE(this->trajectory->get_joint_names().size(), 0);
```

Comment:
@domire8 just doing a final pass check and saw this in my tests which means that we don't throw if the trajectory is empty but we ask for the joint names. While this is a bit inconsistent with other getters, in practice we do keep the the list elements for future additions.

What do you think? Is that ok as is for you? I think it is reasonable for this one.

Reply:
It's consistent with the `JointState` class so yeah its ok